crates/next-core/js/src/internal/page-server-handler.tsx

```diff
@@ -290,6 +299,26 @@ export default function startHandler({
   }
 }
 
+function createNotFoundResponse(isDataReq: boolean): IpcOutgoingMessage {
+  if (isDataReq) {
+    return {
+      type: "response",
+      // Returning a 404 status code is required for the client-side router
+      // to redirect to the error page.
+      statusCode: 404,
+      body: '{"notFound":true}',
+      headers: [["Content-Type", MIME_APPLICATION_JAVASCRIPT]],
```

Comment:
Should this be JSON?

Reply:
The definition is
```rust
headers: Vec<(String, String)>,`
```

Reply:
I mean the `MIME_APPLICATION_JAVASCRIPT`, should that be returning JSON content-type instead?

Reply:
Yes indeed it should. I'll fix this in my `basePath` PR since it also touches this part of the code.